### ISSUE TITLE
Internal error given on attempt to make cases for (bogus) function in let-expression

### ISSUE THREAD

**Issue by mcoblenz** [agda/agda#3591]
In the agda emacs mode, if I put the insertion point in the first hole of the code below and press C-c C-c, I get:
```
An internal error has occurred. Please report this as a bug.
Location of the error: src/full/Agda/Interaction/MakeCase.hs:134
```

I now know that I can't do pattern matching inside a let-expression, but I didn't realize that initially. I suggest that the new error message explicitly suggest using `where` instead -- and eventually, if possible, extend `let` to permit defining functions by pattern matching.

```
open import Data.Nat

foo : ℕ → ℕ
foo n =
  let
    bad : ℕ → ℕ
    bad n = {!!}
  in {!!}
```

**Comment by gallais** [comment]
I think this is #3501 which is fixed in 2.6.0.

**Comment by mcoblenz** [comment]
Oh good, thanks.
